### DISCUSSION TITLE
Using a SFC as a node

### DISCUSSION THREAD

**Discussion by djeetee** [newcat/baklavajs#258] [Q&A]
Hi everyone, 

I would like to know if it is possible to use a Single File Component as a node. The docs indicate that it is possible however the steps are not very clear.

Some background: 

1. I'm using baklava v2 with vue3

2. The structure of my nodes is fairly complex in the senses that the SFC nodes (there will be a few varieties) will display some images and will also allow the use to input text, etc.. 

3. I also would like to be able to connect these nodes to be connected to other nodes (no need for automatic passing of data as long as I can figure out a mechanism to do so)

I'm trying to make sense of what this [page](https://v2.baklava.tech/nodes/interfaces.html#display-component) is saying but I can't figure out the steps to do so: I have a skeleton SFC, how do I register it so that it could be added to the graph and then allow it to be connected to other nodes?

I'm hoping someone could point me in the right direction.

thank you.

**Comment by newcat** [comment]
Sorry for the late reply, I hope this documentation page helps: https://v2.baklava.tech/visual-editor/customization.html#custom-components

Regarding the connecting of nodes: If you use the default `NodeInterface` components inside your node, you should be fine. Otherwise, you basically have to implement the functionality of the `NodeInterface` component yourself as well ([reference](https://github.com/newcat/baklavajs/blob/dev-v2.0/packages/renderer-vue/src/node/NodeInterface.vue))
